### DISCUSSION TITLE
Using the Energy Storage Model with the Grid-Forming Inverter Model

### DISCUSSION THREAD

**Discussion by amparocorvalan** [CURENT/andes#553] [Modeling] [answered]
Dear Dr. Cui and Jinning,, 

For the system I want to model I would like to see the effect of controlling an energy storage device with grid-forming mode. Is there any way to do this using the models that exist? How could I connect the energy storage model with the grid-forming inverter?

Thank you in advance.

Best regards,
Amparo

**Comment by jinningwang** [comment, marked as answer]
As you have participated this discussion, https://github.com/CURENT/andes/discussions/502, the desired "energy storage with grid-forming converter" is not yet available.

A quick implementation can be that, extend the existing grid-forming converter model with energy storage function. Following comments for your reference:
1. In v1.9.1, grid forming converters include `REGCV1`, `REGCV2`, `REGCF1`, `REGCF2`, and `REGCF3`.
1. Development from `PVD1` to `ESD1` can be a good reference for energy storage function development.
